Source/ACE/Entity/Player.cs

```diff
@@ -1122,13 +1139,53 @@ public void HandleActionBuyStartTransaction(WorldObject vendor, List<WorldObject
             }).EnqueueChain();
         }
 
+        public void HandleActionBuyFinalTransaction(WorldObject vendor, List<WorldObject> purchaselist, bool valid)
+        {
+            new ActionChain(this, () =>
+            {
+                if (valid)
+                {
+                    if (CoinValue - escoCoin >= 0)
+                    {
+                        // this is there the money is spent!
+                        if (SpendCoin(escoCoin))
+                        {
+                            escoCoin = 0;
+                            foreach (WorldObject wo in purchaselist)
+                            {
+                                // todo: check for inventory space.
+                                AddNewItemToInventory(wo.WeenieClassId);
```

Comment:
this is fine for weenie-based purchases (heal kits, crafting supplies, etc), but is not for repurchasing items that were looted and then sold to the vendor.